wallet-new/server/Main.hs

```diff
@@ -82,8 +81,8 @@ actionWithWallet sscParams nodeParams ntpConfig wArgs@WalletBackendParams {..} =
         syncWallets
 
     runNodeWithInit ntpStatus init' nr =
-        let (ActionSpec f, outs) = runNode nr (plugins ntpStatus)
-         in (ActionSpec $ \s -> init' >> f s, outs)
+        let f = runNode nr (plugins ntpStatus)
+         in \s -> init' >> f s
```

Comment:
I'd be tempted to simplify further to 

```haskell
runNodeWithInit ntpStatus init' nr s = do
  init'
  runNode nr (plugins ntpStatus) s
```

but this is already a big improvement :)